### ISSUE TITLE
Wrong number of results is returned when LIMIT is used with intersecting indexes

### ISSUE THREAD

**Issue by tobiemh** [orientechnologies/orientdb#3627]
```sql
CREATE CLASS Person EXTENDS V
CREATE CLASS Account EXTENDS V
CREATE PROPERTY Person.account LINK Account
CREATE PROPERTY Person.firstname STRING
CREATE PROPERTY Person.lastname STRING
CREATE INDEX Person.account NOTUNIQUE_HASH_INDEX
CREATE INDEX Person.firstname NOTUNIQUE_HASH_INDEX
CREATE INDEX Person.lastname NOTUNIQUE_HASH_INDEX
```

The following query has a LIMIT but the wrong number of results is returned...

```sql
EXPLAIN select from Person where account=#17:106 OR firstname='Tobie' OR lastname='Williams' LIMIT 100
```
```json
{
    "result": [
        {
            "limit": 100,
            "compositeIndexUsed": 3,
            "involvedIndexes": ["Person.lastname","Person.account","Person.firstname"],
            "evaluated": 103,
            "documentReads": 103,
            "documentAnalyzedCompatibleClass": 103,
            "recordReads": 103,
            "resultSize": 103,
        }
    ],
}
```
This same problem can also be seen in the example at the top of issue #3626 

***
OrientDB version: 2.0.2

**Comment by thejeed** [comment]
Same or comparable Problem here with 2.0.5:
SELECT FROM city WHERE localName CONTAINSTEXT 'Spitz' LIMIT 100; -> 100 results
SELECT FROM city WHERE localName CONTAINSTEXT 'Spitz' OR readableName CONTAINSTEXT 'Spitz' LIMIT 100; -> 101 results

localName and readableName each have its own FULLTEXT index.

**Comment by luigidellaquila** [comment]
fixed in 2.0.x branch, will be released with 2.0.6
